level-ws.js

```diff
@@ -69,62 +73,58 @@ inherits(WriteStream, Writable)
 
 WriteStream.prototype._write = function write (d, enc, next) {
   var self = this
+  // TODO use self.destroyed
   if (self._destroyed) return
 
+  // TODO remove, no longer needed
   if (!self._db.isOpen()) {
     return self._db.once('ready', function () {
       write.call(self, d, enc, next)
     })
   }
 
-  if (self._options.maxBufferLength &&
-      self._buffer.length > self._options.maxBufferLength) {
-    self.once('_flush', next)
-  } else {
-    if (self._buffer.length === 0) {
-      process.nextTick(function () { self._flush() })
-    }
-    self._buffer.push(d)
-    next()
-  }
+  // TODO re-implement buffering logic using this._options.maxBufferLength
+  self._buffer.push(d)
+  next()
 }
 
-WriteStream.prototype._flush = function (f) {
+WriteStream.prototype._final = function (cb) {
```

Comment:
According to documentation, implementations of `WriteStream` should implement `_write()` and optionally `_final()`.